virtcontainers/store/filesystem.go

```diff
@@ -42,13 +128,53 @@ func (f *filesystem) trace(name string) (opentracing.Span, context.Context) {
 	return span, ctx
 }
 
+func (f *filesystem) itemToPath(item Item) (string, error) {
+	fileName, err := itemToFile(item)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(f.path, fileName), nil
+}
+
+func (f *filesystem) initialize() error {
+	_, err := os.Stat(f.path)
+	if err == nil {
+		return nil
+	}
+
+	// Our root path does not exist, we need to create the initial layout:
+	// The root directory and a lock file
+
+	// Root directory
+	if err := os.MkdirAll(f.path, DirMode); err != nil {
+		return err
+	}
+
+	// Lock
+	lockPath := filepath.Join(f.path, LockFile)
+
+	lockFile, err := os.Create(lockPath)
```

Comment:
Just trying to follow here. We're creating a lock file for every store, and we have one store per type of root path such as `/var/run/` or `var/lib`, right?

I'm just concerned that we might create too many lock files if we cannot control that too many stores have been created.